packages/react/src/jsx/ReactJSXElement.js

```diff
@@ -380,8 +403,16 @@ export function jsxProdSignatureRunningInDevWithDynamicChildren(
       isStaticChildren,
       source,
       self,
-      __DEV__ && Error('react-stack-top-frame'),
-      __DEV__ && createTask(getTaskName(type)),
+      // TODO: Reuse a placeholder stack to indicate the deopt.
+      __DEV__ &&
+        (recentlyCreatedOwnerStacks++ < ownerStackLimit
+          ? Error('react-stack-top-frame')
+          : unknownOwnerDebugStack),
+      // TODO: If task.run is not the bottleneck, reuse a placeholder task to indicate the deop.
+      __DEV__ &&
+        (recentlyCreatedDebugTasks++ < debugTaskLimit
```

Comment:
This is the way.

Almost minimum overhead per call. However, what's the point of having two counters that should always line up? It would be even more minimum if it was just incrementing once and comparing twice.